ts/features/pushNotifications/utils/configurePushNotification.ts

```diff
@@ -69,80 +79,107 @@ const onPushNotificationTokenAvailable = (
 
   const state = store.getState();
   void updateMixpanelProfileProperties(state);
 };
 
-/**
- * Helper type used to validate the notification payload.
- * The message_id can be in different places depending on the platform.
- */
-const NotificationPayload = t.partial({
-  message_id: NonEmptyString,
-  data: t.partial({
-    message_id: NonEmptyString
-  })
-});
-
 const onPushNotificationReceived = (
   notification: Omit<ReceivedNotification, "userInfo">,
   store: Store
-) =>
-  pipe(
-    notification.userInteraction,
-    B.fold(
-      () => E.left(undefined),
-      () =>
-        pipe(
-          notification,
-          NotificationPayload.decode,
-          E.mapLeft(trackMessageNotificationParsingFailure)
-        )
-    ),
-    O.fromEither,
-    O.chain(payload =>
-      pipe(
-        payload.message_id,
-        O.fromNullable,
-        O.alt(() =>
-          pipe(
-            payload.data,
-            O.fromNullable,
-            O.chainNullableK(_ => _.message_id)
-          )
-        )
-      )
-    ),
-    // We just received a push notification about a new message
-    O.map(messageId =>
-      pipe(trackMessageNotificationTap(messageId), trackingResult =>
-        pipe(
-          notification.foreground,
-          B.foldW(
-            // The App was closed/in background and has been now opened clicking
-            // on the push notification.
-            // Save the message id of the notification in the store so the App can
-            // navigate to the message detail screen as soon as possible (if
-            // needed after the user login/insert the unlock code)
-            () =>
-              store.dispatch(
-                updateNotificationsPendingMessage({
-                  id: messageId,
-                  foreground: false,
-                  trackEvent: trackingResult === undefined
-                })
-              ),
-            // The App is in foreground so just refresh the messages list
-            () => handleForegroundMessageReload(store)
-          )
-        )
-      )
-    ),
-    // On iOS we need to call this when the remote notification handling is complete
-    () => notification.finish(PushNotificationIOS.FetchResult.NoData)
-  );
+) => {
+  const userOptedInForAnalytics = hasUserOptedInForAnalytics(store.getState());
+  if (notification.userInteraction) {
+    const messageId = messageIdFromPushNotification(
+      notification,
+      userOptedInForAnalytics
+    );
+    if (messageId != null) {
+      handleMessagePushNotification(
+        notification.foreground,
+        messageId,
+        store,
+        userOptedInForAnalytics
+      );
+    }
+  }
+
+  // Signal the system that the notification has been processed (this
+  // is mandatory even if there was an errore. Failing to do so can
+  // lead the system to not deliver push notifications anymore)
+  notification.finish(PushNotificationIOS.FetchResult.NoData);
+};
+
+const messageIdFromPushNotification = (
```

Comment:
this function returns either `undefined` or a nullish id in case of error.
I think that returning undefined in case of nullish id instead of leaving it to the runtime would increase stability, and avoid a runtime failure on a check like `messageId===undefined`, while TSC would not detect it (`return undefined` on line 150 should do it)
this is also because TSC reports a return type like `(string & INonEmptyStringTag) | undefined`

Reply:
Done in ba4f8e84b28e6473b63ed096e714800bba43ad8a